### PR TITLE
KPIS-462_add-new-max-value-function-into-istore

### PR DESCRIPTION
**Pull request by m7shapan**
#### [Ticket](https://adjustcom.atlassian.net/browse/KPIS-462)
update `istore` extension to add `max_value` function

### CODE REVIEW

File: src/istore.c [hunk 1]
```diff
@@ -167,6 +167,39 @@ Datum istore_max_key(PG_FUNCTION_ARGS)
...
+
+    is    = PG_GETARG_IS(0);
+    if (is->len == 0)
+    {
+        PG_RETURN_NULL();
+    }
+
+    index = 0;
+    pairs = FIRST_PAIR(is, IStorePair);
+    value = pairs[index].val;
```
Comment (za-arthur) [inline src/istore.c:189]: Just little optimization. You probably don't need to check first element again in the loop and just use `for` loop:

```suggestion
    pairs = FIRST_PAIR(is, IStorePair);
    value = pairs[0].val;
    for (int i = 1; i < is->len; i++)
```

Comment (m7shapan) [reply src/istore.c]: Done

File: src/istore.c [hunk 2]
```diff
@@ -167,6 +167,36 @@ Datum istore_max_key(PG_FUNCTION_ARGS)
...
+
+    pairs = FIRST_PAIR(is, IStorePair);
+    value = pairs[0].val;
+    for (int i = 1; i < is->len; i++)
+    {
+        if (value < pairs[i].val)
+        {
+            value = pairs[i].val;
+        }
+
```
Comment (za-arthur) [inline src/istore.c:194]:
```suggestion
        if (value < pairs[i].val)
            value = pairs[i].val;
```

**Review by za-arthur — APPROVED** [review]
Thanks! Looks good to me.

I just left couple suggestions. But it also would be good to rename test file `functions_max_value_test.sql` to something generic like `functions_test.sql`.

This test file is not generated. Which I think is good. Personally I think test generation functionality here is unnecessary, it adds unnecessary complexity and it makes tests very slow because of many `BEGIN`, `CREATE EXTENSION istore` statements:

For example `functions_plain_test` runs for 13 seconds on my laptop:
```
...
test functions_plain_test         ... ok        13940 ms
...
```

After I removed all that transactions and `CREATE EXTENSION`:
```
...
test functions_plain_test         ... ok          182 ms
...
```

13 sec vs 182 ms.
But this PR is not about fixing existing tests. We can try to work on this separately.

**Review by rapimo — APPROVED** [review]
great work!

**Comment by za-arthur** [comment]
I deployed and updated `istore` on backends, shards, demo servers and esh-pg-tracker servers.

**Pull request merged**